File: Reverse Words in a String.java
class Solution {
    public String reverseWords(String s) {

        if (s.length() == 0) return s ;

        int flag = 0  ;

        for ( int i = 0 ; i < s.length() ; i++ ) {
            if (s.charAt(i) != ' ') {
                flag = 1 ;
                break ;
            }
        }

        if (flag == 0) return "" ;

        String[] words = s.split(" ") ;

        String res = "" ;

        Collections.reverse(Arrays.asList(words));

        for ( int i = 0 ; i < words.length ; i++ ) {
            if (words[i].length() > 0) {
                res = res + words[i] + " " ;
            }
        }

        return res.substring(0, res.length() - 1) ;
    }
}

File: Reverse Words in a String.cpp
class Solution {
public:
    string reverseWords(string s) {

        if (s.length() == 0) return s ;

        int flag = 0 ;

        for (int i = 0 ; i < s.length() ; i++) {
            if (s[i] != ' ') {
                flag = 1 ;
                break ;
            }
        }

        if (flag == 0) return "" ;

        vector<string> dummy ;

        stringstream ss(s) ;
        string word ;

        while (getline(ss, word, ' ')) {
            dummy.push_back(word) ;
        }

        vector<string> words ;

        for (string k : dummy) {
            if (k.length() > 0) {
                words.push_back(k) ;
            }
        }

        reverse(words.begin(), words.end()) ;

        string res ;

        for (string k : words) {
            res = res + k + " " ;
        }
        res.pop_back() ;
        return res ;
    }
};